txredisapi.py

```diff
@@ -2114,6 +2142,14 @@ class RedisFactory(protocol.ReconnectingClientFactory):
     maxDelay = 10
     protocol = RedisProtocol
 
+    def clientConnectionFailed(self, connector, reason):
+        if not self.continueTrying:
+            if self.deferred:
+                self.deferred.errback(reason)
+                self.deferred = None
+        else:
+            protocol.ReconnectingClientFactory(self, connector, reason)
```

Comment:
You probably meant `protocol.ReconnectingClientFactory.clientConnectionFailed` call